---
title: "STAT_306_Project_Group_10"
format: pdf
editor: visual
---

# STAT 306 Project (Group 10)

## Introduction

Breast invasive ductal carcinoma (IDC) is the most common type of breast cancer, with about 80% of all forms of breast cancer being IDC, according to the American Cancer Society (DePolo, 2024). There are numerous nonsurgical treatments of IDC, such as radiotherapy, chemotherapy, and hormone therapy (Wright, 2023), but it is unclear how combinations of treatments can interact in a model to predict a patient’s survival until death.

For our research project, we have selected a dataset of approximately 1900 primary breast cancer samples, obtained from the Molecular Taxonomy of Breast Cancer International Consortium (METABRIC) database through cBioPortal for Cancer Genomics.

Our project question is:

**"How do radiotherapy, chemotherapy, and hormone therapy influence the length of time a patient with IDC will survive until death, given control variables age, surgery type, and tumor stage?"**

Our analysis will involve the inference of covariates within linear models, and thus \_\_\_

-   Variables from columns 32 to 693 consist of genetic attributes containing m-RNA levels z-score for 331 genes, and mutation for 175 genes; they have been omitted due to being difficult to interpret.

-   Due to the distribution of “cancer_type_detailed” categories and for ease of computation, we will be filtering the dataset for deceased IDC patients only, as IDC consists of the majority of the dataset, and the “overall_survival” column for living patients would be inconsistent for living patients due to time.

-   “tumor_size” has been removed due to already partly contributing to “tumor_stage”. (Susan G. Komen®, 2024)

## Analysis

### Loading relevant libraries

```{r}
library(dplyr)
library(tidyverse)
library(ggplot2)
# install.packages("fastDummies")
library(fastDummies)
# install.packages('stringi')
```

### Uploading relevant table

-   

```{r}
cancer <- read.csv("METABRIC_RNA_Mutation.csv", header=TRUE)[,1:31]  %>% 
  transform(type_of_breast_surgery = as.factor(type_of_breast_surgery), 
            cancer_type = as.factor(cancer_type), 
            cancer_type_detailed = as.factor(cancer_type_detailed), 
            pam50_._claudin.low_subtype = as.factor(pam50_._claudin.low_subtype), 
            cohort = as.factor(cohort), 
            er_status_measured_by_ihc = as.factor(er_status_measured_by_ihc), 
            er_status = as.factor(er_status), 
            neoplasm_histologic_grade = as.factor(neoplasm_histologic_grade), 
            her2_status_measured_by_snp6 = as.factor(her2_status_measured_by_snp6), 
            her2_status = as.factor(her2_status), 
            tumor_other_histologic_subtype = as.factor(tumor_other_histologic_subtype), 
            inferred_menopausal_state = as.factor(inferred_menopausal_state), 
            integrative_cluster = as.factor(integrative_cluster), 
            primary_tumor_laterality = as.factor(primary_tumor_laterality), 
            oncotree_code = as.factor(oncotree_code), 
            overall_survival = as.factor(overall_survival), 
            pr_status = as.factor(pr_status), 
#            "3-gene_classifier_subtype" = as.factor(3-gene_classifier_subtype), 
            death_from_cancer = as.factor(death_from_cancer), 
            tumor_stage = as.factor(tumor_stage), 
            chemotherapy = as.factor(chemotherapy),
            hormone_therapy = as.factor(hormone_therapy), 
            radio_therapy = as.factor(radio_therapy)
            ) 
cancer
summary(cancer)
```

### Narrowing down to data of interest

```{r}
cancer_new <- cancer %>% 
  drop_na(age_at_diagnosis, type_of_breast_surgery, tumor_size, tumor_stage, chemotherapy, hormone_therapy, radio_therapy, overall_survival_months) %>% 
  filter(!(age_at_diagnosis==""), !(type_of_breast_surgery==""), !(tumor_size==""), !(tumor_stage==""), !(chemotherapy==""), !(hormone_therapy==""), !(radio_therapy==""), !(overall_survival_months==""), !(cancer_type_detailed==""), overall_survival == 1, !(tumor_stage==0)) %>% 
  transform(
    type_of_breast_surgery = as.factor(type_of_breast_surgery), 
    tumor_stage = as.factor(tumor_stage), 
    chemotherapy = as.factor(chemotherapy), 
    hormone_therapy = as.factor(hormone_therapy), 
    radio_therapy = as.factor(radio_therapy)
    ) %>% 
  select(age_at_diagnosis, type_of_breast_surgery, cancer_type_detailed, tumor_stage, chemotherapy, hormone_therapy, radio_therapy, overall_survival_months) 

cancer_new$tumor_stage <- relevel(factor(cancer_new$tumor_stage), ref = "1")
cancer_new$cancer_type_IDC <- (cancer_new$cancer_type_detailed == "Breast Invasive Ductal Carcinoma") 

head(cancer_new, 10)
summary(cancer_new)
```

```{r}
boxplot(overall_survival_months~cancer_type_IDC, data=cancer_new)
```

### Exploratory data analysis

```{r}
boxplot(overall_survival_months~type_of_breast_surgery, data=cancer_new)
```

```{r}
unique(cancer$pam50_._claudin.low_subtype)
unique(cancer$her2_status_measured_by_snp6)
unique(cancer$oncotree_code)
unique(cancer$tumor_stage)
```

```{r}
cancer_new <- cancer %>% 
  filter(!is.na(age_at_diagnosis), !is.na(tumor_size), !is.na(tumor_stage), !is.na(chemotherapy), !is.na(hormone_therapy), !is.na(radio_therapy), !is.na(overall_survival_months)) %>% 
  select(age_at_diagnosis, tumor_size, tumor_stage, chemotherapy, hormone_therapy, radio_therapy, overall_survival_months) 
# %>% 
#   dummy_cols(select_columns = c("tumor_stage","chemotherapy", "hormone_therapy", "radio_therapy"), remove_selected_columns = TRUE, remove_first_dummy  = TRUE)
cancer_new
```

```{r}
cancer_new_2 <- cancer %>% 
  filter(!is.na(age_at_diagnosis), !is.na(type_of_breast_surgery), !is.na(tumor_size), !is.na(tumor_stage), !is.na(chemotherapy), !is.na(hormone_therapy), !is.na(radio_therapy), !is.na(overall_survival_months), cancer_type_detailed == "Breast Invasive Ductal Carcinoma", overall_survival == 1) %>% 
  select(age_at_diagnosis, type_of_breast_surgery, tumor_stage, chemotherapy, hormone_therapy, radio_therapy, overall_survival_months) 
cancer_new_2
```

```{r}
options(repr.plot.width = 8, repr.plot.height = 5)
# Create a scatter plot with a regression line
ggplot(cancer_new, aes(x = age_at_diagnosis, y = overall_survival_months)) +
  geom_point() +  # Scatter plot points
#  geom_smooth(method = "lm", formula = verall_survival_months ~ age_at_diagnosis, se = TRUE, color = "blue", na.rm = TRUE) +  # Regression line with confidence interval
  geom_smooth(method = "lm", se = TRUE, color = "blue", na.rm = TRUE) +  # Regression line with confidence interval
  theme(
    text = element_text(size = 10), 
    plot.title = element_text(face = "bold"), 
    axis.title = element_text(face = "bold") 
  ) +
  labs(title = "Overall Survival against Age",
       x = "Age (Years)",
       y = "Overall Survival (Months)")
```

```{r}
fullmod = lm(overall_survival_months ~ .*.,data=cancer_new_2)
step(fullmod, direction = "both")
```
