---
title: "STAT_306_Project_Group_10"
format: pdf
editor: visual
execute:
  echo: false # Hide code in document
---

# STAT 306 Project

## Nicholas Tam, Ivy Cheung, Kaichi Nakajima, Kevin Liu (Group 10)

## Introduction

Breast invasive ductal carcinoma (IDC) is the most common type of breast cancer, with about 80% of all forms of breast cancer being IDC, according to the American Cancer Society (DePolo, 2024). There are numerous nonsurgical treatments of IDC, such as radiotherapy, chemotherapy, and hormone therapy (Wright, 2023), but it is unclear how combinations of treatments can interact in a model to predict a patient’s survival until death.

For our research project, we have selected a dataset of approximately 1900 primary breast cancer samples, obtained from the Molecular Taxonomy of Breast Cancer International Consortium (METABRIC) database through cBioPortal for Cancer Genomics.

Our project question is:

**"How do radiotherapy, chemotherapy, and hormone therapy influence the length of time a patient with IDC will survive until death, given control variables age, surgery type, and tumor stage?"**

Our analysis will involve the inference of covariates within linear models, and thus \_\_\_

-   Variables from columns 32 to 693 consist of genetic attributes containing m-RNA levels z-score for 331 genes, and mutation for 175 genes; they have been omitted due to being difficult to interpret.

## Analysis

### Loading relevant libraries

```{r}
library(dplyr) # Data manipulation operations
library(tidyverse) # Better presentation of data
library(ggplot2) # Provides commands to create complex plots
# install.packages("fastDummies")
library(fastDummies) 
# install.packages('stringi')
library(car) # Applied regression tools, including VIF
# install.packages('gridExtra')
library(gridExtra) # Extensions for grid system
```

### Uploading relevant table and cleaning data

-   Due to the distribution of "cancer_type_detailed" categories and for ease of computation, we will be filtering the dataset for deceased IDC patients only, as IDC consists of the majority of the dataset, and the "overall_survival" column for living patients would be inconsistent for living patients due to time.

-   "tumor_size" has been removed due to already partly contributing to "tumor_stage". (Susan G. Komen®, 2024)

-   All rows with "N/A" or "" as their values for any column have been removed to ensure no data is missing.

-   All categorical variables are transformed to be treated as categorical by the table.

-   After initial "tumor_stage==0" and "tumor_stage==4" have been removed due to lack of sufficient amounts of data points.

```{r}
cancer = read.csv("METABRIC_RNA_Mutation.csv", header=TRUE)[,1:31]
cancer_new_1 <- cancer %>% 
  drop_na(age_at_diagnosis, type_of_breast_surgery, tumor_size, tumor_stage, chemotherapy, hormone_therapy, radio_therapy, overall_survival_months, death_from_cancer) %>% 
  filter(!(age_at_diagnosis==""), !(type_of_breast_surgery==""), !(tumor_size==""), !(tumor_stage==""), !(chemotherapy==""), !(hormone_therapy==""), !(radio_therapy==""), !(overall_survival_months==""), !(death_from_cancer==""), cancer_type_detailed == "Breast Invasive Ductal Carcinoma",) %>% 
  transform(
    type_of_breast_surgery = as.factor(type_of_breast_surgery), 
    tumor_stage = as.factor(tumor_stage), 
    chemotherapy = as.factor(chemotherapy), 
    hormone_therapy = as.factor(hormone_therapy), 
    radio_therapy = as.factor(radio_therapy), 
    death_from_cancer = as.factor(death_from_cancer)
    ) %>% 
  select(age_at_diagnosis, type_of_breast_surgery, tumor_stage, chemotherapy, hormone_therapy, radio_therapy, death_from_cancer,  overall_survival_months) 
# !(death_from_cancer=="")
summary(cancer_new_1)

cancer_new <- cancer_new_1 %>% 
  filter(!(tumor_stage==0), !(tumor_stage==4),)
  
cancer_new$tumor_stage <- relevel(factor(cancer_new$tumor_stage), ref = "1")
cancer_new$death_from_cancer <- relevel(factor(cancer_new$death_from_cancer), ref = "Living")

head(cancer_new, 10)

summary(cancer_new)
```

### Exploratory data analysis

#### Residual plot and QQ-plot

-   There does not appear to be any observable patterns in residual values beyond being centered around 0.

-   QQ-plot indicates distribution of residuals is light-tailed normal.

```{r}
reg_start = lm(overall_survival_months~age_at_diagnosis,data=cancer_new)
plot(reg_start$fitted.values, reg_start$residuals, xlab="Fitted Values", ylab="Residuals")
abline(a = 0, b = 0)
lines(lowess(reg_start$fitted.values, reg_start$residuals), col = "blue", lwd = 2)
qqnorm(reg_start$residuals)
qqline(reg_start$residuals)
```

#### Boxplots

-   

```{r}
box_age_type <- cancer_new %>%
    ggplot() +
    geom_boxplot(aes(type_of_breast_surgery, overall_survival_months, fill = type_of_breast_surgery)) +
    theme(
        text = element_text(size = 8),
        plot.title = element_text(face = "bold"),
        axis.title = element_text(face = "bold"), 
        axis.text.x = element_text(angle = 45)
    ) +
    ggtitle("Age Comparison") +
    labs(title = "Overall Survival for each Breast Surgery",
       x = "Age (Years)",
       y = "Overall Survival (Months)", 
       color = "Breast Surgery") + 
    stat_summary(aes(type_of_breast_surgery, overall_survival_months, fill = type_of_breast_surgery),
                 fun = mean, colour = "yellow", geom = "point", shape = 18, size = 5
                )

box_age_chemo <- cancer_new %>%
    ggplot() +
    geom_boxplot(aes(chemotherapy, overall_survival_months, fill = chemotherapy)) +
    theme(
        text = element_text(size = 8),
        plot.title = element_text(face = "bold"),
        axis.title = element_text(face = "bold"), 
#        axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)
    ) +
    ggtitle("Age Comparison") +
    labs(title = "Overall Survival for Chemotherapy",
       x = "Age (Years)",
       y = "Overall Survival (Months)", 
       color = "Chemotherapy") + 
    stat_summary(aes(chemotherapy, overall_survival_months, fill = chemotherapy),
                 fun = mean, colour = "yellow", geom = "point", shape = 18, size = 5
                )

box_age_hormo <- cancer_new %>%
    ggplot() +
    geom_boxplot(aes(hormone_therapy, overall_survival_months, fill = hormone_therapy)) +
    theme(
        text = element_text(size = 8),
        plot.title = element_text(face = "bold"),
        axis.title = element_text(face = "bold"), 
#        axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)
    ) +
    ggtitle("Age Comparison") +
    labs(title = "Overall Survival for Hormone Therapy",
       x = "Age (Years)",
       y = "Overall Survival (Months)", 
       color = "Hormone Therapy") + 
    stat_summary(aes(hormone_therapy, overall_survival_months, fill = hormone_therapy),
                 fun = mean, colour = "yellow", geom = "point", shape = 18, size = 5
                )

box_age_radio <- cancer_new %>%
    ggplot() +
    geom_boxplot(aes(radio_therapy, overall_survival_months, fill = radio_therapy)) +
    theme(
        text = element_text(size = 8),
        plot.title = element_text(face = "bold"),
        axis.title = element_text(face = "bold"), 
#        axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)
    ) +
    ggtitle("Age Comparison") +
    labs(title = "Overall Survival for Radio Therapy",
       x = "Age (Years)",
       y = "Overall Survival (Months)", 
       color = "Radio Therapy") + 
    stat_summary(aes(radio_therapy, overall_survival_months, fill = radio_therapy),
                 fun = mean, colour = "yellow", geom = "point", shape = 18, size = 5
                )

box_age_tumor <- cancer_new %>%
    ggplot() +
    geom_boxplot(aes(tumor_stage, overall_survival_months, fill = tumor_stage)) +
    theme(
        text = element_text(size = 8),
        plot.title = element_text(face = "bold"),
        axis.title = element_text(face = "bold"), 
#        axis.text.x = element_text(angle = 45)
    ) +
    ggtitle("Age Comparison") +
    labs(title = "Overall Survival for Tumor Stage",
       x = "Age (Years)",
       y = "Overall Survival (Months)", 
       color = "Tumor Stage") + 
    stat_summary(aes(tumor_stage, overall_survival_months, fill = tumor_stage),
                 fun = mean, colour = "yellow", geom = "point", shape = 18, size = 5
                )

box_age_death <- cancer_new %>%
    ggplot() +
    geom_boxplot(aes(death_from_cancer, overall_survival_months, fill = death_from_cancer)) +
    theme(
        text = element_text(size = 8),
        plot.title = element_text(face = "bold"),
        axis.title = element_text(face = "bold"), 
        axis.text.x = element_text(angle = 45)
    ) +
    ggtitle("Age Comparison") +
    labs(title = "Overall Survival for Living Condition",
       x = "Age (Years)",
       y = "Overall Survival (Months)", 
       color = "Living Condition") + 
    stat_summary(aes(death_from_cancer, overall_survival_months, fill = death_from_cancer),
                 fun = mean, colour = "yellow", geom = "point", shape = 18, size = 5
                )

grid.arrange(box_age_chemo, box_age_hormo, nrow = 1)
grid.arrange(box_age_radio, box_age_type, nrow = 1)
grid.arrange(box_age_tumor, box_age_death, nrow = 1)
```

#### Scatterplots of survival against interactions between age and categorical variables 

-   

```{r}
options(repr.plot.width = 20, repr.plot.height = 5)

reg_0 = lm(overall_survival_months~age_at_diagnosis*type_of_breast_surgery,data=cancer_new)  # TODO
summary(reg_0)
plot_age_type <- ggplot(cancer_new, aes(x = age_at_diagnosis, y = overall_survival_months, color = type_of_breast_surgery)) +
  geom_point() +  
  geom_smooth(method = "lm", na.rm = TRUE) +  
  theme(
    text = element_text(size = 7), 
    plot.title = element_text(face = "bold"), 
    axis.title = element_text(face = "bold") 
  ) +
  labs(title = "Overall Survival against Age and Breast Surgery",
       x = "Age (Years)",
       y = "Overall Survival (Months)", 
       color = "Breast Surgery")

reg_1 = lm(overall_survival_months~age_at_diagnosis*chemotherapy,data=cancer_new)  # TODO
summary(reg_1)
plot_age_chemo <- ggplot(cancer_new, aes(x = age_at_diagnosis, y = overall_survival_months, color = chemotherapy)) +
  geom_point() +  
  geom_smooth(method = "lm", na.rm = TRUE) +  
  theme(
    text = element_text(size = 7), 
    plot.title = element_text(face = "bold"), 
    axis.title = element_text(face = "bold") 
  ) +
  labs(title = "Overall Survival against Age and Chemotherapy",
       x = "Age (Years)",
       y = "Overall Survival (Months)", 
       color = "Chemotherapy")

reg_2 = lm(overall_survival_months~age_at_diagnosis*hormone_therapy,data=cancer_new)  # TODO
summary(reg_2)
plot_age_hormo <- ggplot(cancer_new, aes(x = age_at_diagnosis, y = overall_survival_months, color = hormone_therapy)) +
  geom_point() +  
  geom_smooth(method = "lm", na.rm = TRUE) +  
  theme(
    text = element_text(size = 7), 
    plot.title = element_text(face = "bold"), 
    axis.title = element_text(face = "bold") 
  ) +
  labs(title = "Overall Survival against Age and Hormone Therapy",
       x = "Age (Years)",
       y = "Overall Survival (Months)", 
       color = "Hormone Therapy")

reg_3 = lm(overall_survival_months~age_at_diagnosis*radio_therapy,data=cancer_new)  # TODO
summary(reg_3)
plot_age_radio <- ggplot(cancer_new, aes(x = age_at_diagnosis, y = overall_survival_months, color = radio_therapy)) +
  geom_point() +  
  geom_smooth(method = "lm", na.rm = TRUE) +  
  theme(
    text = element_text(size = 7), 
    plot.title = element_text(face = "bold"), 
    axis.title = element_text(face = "bold") 
  ) +
  labs(title = "Overall Survival against Age and Radio Therapy",
       x = "Age (Years)",
       y = "Overall Survival (Months)", 
       color = "Radio Therapy")

reg_4 = lm(overall_survival_months~age_at_diagnosis*tumor_stage,data=cancer_new)  # TODO
summary(reg_4)
plot_age_tumor <- ggplot(cancer_new, aes(x = age_at_diagnosis, y = overall_survival_months, color = tumor_stage)) +
  geom_point() +  
  geom_smooth(method = "lm", na.rm = TRUE) +  
  theme(
    text = element_text(size = 7), 
    plot.title = element_text(face = "bold"), 
    axis.title = element_text(face = "bold") 
  ) +
  labs(title = "Overall Survival against Age and Tumor Stage",
       x = "Age (Years)",
       y = "Overall Survival (Months)", 
       color = "Tumor Stage")

# reg_3_1 = lm(overall_survival_months~age_at_diagnosis+radio_therapy,data=cancer_new)  # TODO
# summary(reg_3_1)
# scatterplot(overall_survival_months~age_at_diagnosis+radio_therapy, data=cancer_new, smooth=FALSE, main="Overall Survival against Age and Radio Therapy", xlab="Age (Years)", ylab="Overall Survival (Months)")

reg_5 = lm(overall_survival_months~age_at_diagnosis*death_from_cancer,data=cancer_new)  # TODO
summary(reg_5)
plot_age_death <- ggplot(cancer_new, aes(x = age_at_diagnosis, y = overall_survival_months, color = death_from_cancer)) +
  geom_point() +  
  geom_smooth(method = "lm", na.rm = TRUE) +  
  theme(
    text = element_text(size = 7), 
    plot.title = element_text(face = "bold"), 
    axis.title = element_text(face = "bold") 
  ) +
  labs(title = "Overall Survival against Age and Living Condition",
       x = "Age (Years)",
       y = "Overall Survival (Months)", 
       color = "Living Condition")

# grid.arrange(plot_age_chemo, plot_age_hormo, plot_age_radio)
# grid.arrange(plot_age_type, plot_age_tumor, plot_age_death)

# grid.arrange(plot_age_chemo, plot_age_hormo, nrow = 1)
# grid.arrange(plot_age_radio, plot_age_type, nrow = 1)
# grid.arrange(plot_age_tumor, plot_age_death, nrow = 1)

grid.arrange(plot_age_chemo, plot_age_hormo, ncol = 1)
grid.arrange(plot_age_radio, plot_age_type, ncol = 1)
grid.arrange(plot_age_tumor, plot_age_death, ncol = 1)

# plot_age_chemo
# plot_age_hormo
# plot_age_radio
# plot_age_type
# plot_age_tumor
# plot_age_death
```

### Model selection

-   The interaction terms for all therapies

```{r}
# fullmod = lm(overall_survival_months ~ .*.,data=cancer_new)
fullformula = overall_survival_months ~ age_at_diagnosis + chemotherapy * hormone_therapy * radio_therapy + age_at_diagnosis : chemotherapy + age_at_diagnosis : hormone_therapy + age_at_diagnosis : radio_therapy + tumor_stage + tumor_stage : chemotherapy + tumor_stage : hormone_therapy + tumor_stage : radio_therapy + type_of_breast_surgery + death_from_cancer

fullmod = lm(fullformula, data=cancer_new)
# finalmod = step(fullmod, direction = "both", trace = 1)
finalmod_both = step(fullmod, direction = "both", trace = 0)
summary(finalmod_both)
```

```{r}
finalmod_backward = step(fullmod, direction = "backward", trace = 0)
summary(finalmod_backward)
```

```{r}
finalmod_forward = step(fullmod, direction = "forward", trace = 0)
summary(finalmod_back)
```

```{r}
# # Define the function for backward selection
# backward_selection <- function(data, formula) {
#   # Fit the full model
#   models <- list()
#   models[[1]] <- lm(formula, data = data)
#   # print(summary(models[[1]]))
#   
#   # Perform backward selection
#   current_formula <- formula
#   model_index <- 2
#   
#   while (TRUE) {
#     model <- lm(current_formula, data = data)
#     
#     # Get p-values of the coefficients
#     p_values <- summary(model)$coefficients[, 4]
#     
#     # Exclude the intercept
#     p_values <- p_values[-1]
#     
#     # Check if any predictor has a p-value greater than 0.05
#     if (max(p_values, na.rm = TRUE) > 0.05) {
#       # Remove the predictor with the highest p-value
#       predictor_to_remove <- names(which.max(p_values))
#       current_formula <- update(current_formula, as.formula(paste(". ~ . -", predictor_to_remove)))
#       
#       # Fit the new model
#       models[[model_index]] <- lm(current_formula, data = data)
#       # print(summary(models[[model_index]]))
#       model_index <- model_index + 1
#     } else {
#       # Stop if all predictors are significant
#       break
#     }
#   }
#   
#   return(models)
# }
# 
# finalmod2 = backward_selection(cancer_new, fullformula)
```

```{r}
# Create the residual plot
# TODO
plot(finalmod_both$fitted.values, finalmod_both$residuals, xlab="Fitted Values", ylab="Residuals")
abline(a = 0, b = 0)
lines(lowess(finalmod_both$fitted.values, finalmod_both$residuals), col = "blue", lwd = 2)
```

```{r}
qqnorm(finalmod_both$residuals)
qqline(finalmod_both$residuals)
```

## Conclusion

## Appendix

-   Breast cancer gene expression profiles (METABRIC). Kaggle. (2016, May 10). <https://www.kaggle.com/datasets/raghadalharbi/breast-cancer-gene-expression-profiles-metabric>   

-   CBioPortal for Cancer Genomics. (n.d.). <https://www.cbioportal.org/study/summary?id=brca_metabric>

-   DePolo, J. (2024, October 2). Invasive ductal carcinoma (IDC). Breastcancer.org - Breast Cancer Information and Support. <https://www.breastcancer.org/types/invasive-ductal-carcinoma>  

-   Wright , P. (2023, March 21). Invasive ductal carcinoma (IDC). Johns Hopkins Medicine. <https://www.hopkinsmedicine.org/health/conditions-and-diseases/breast-cancer/invasive-ductal-carcinoma-idc#:~:text=Radiation%20therapy%20might%20be%20part,lymph%20nodes%2C%E2%80%9D%20Wright%20says>. 

-   Tumor size and staging. Susan G. Komen®. (2024, May 2). <https://www.komen.org/breast-cancer/diagnosis/stages-staging/tumor-size/#:~:text=Tumor%20size%20is%20related%20to,the%20size%20of%20the%20tumor>.
